File: example/src/main/java/com/vmloft/develop/library/example/demo/custom/CustomViewActivity.java
package com.vmloft.develop.library.example.demo.custom;

import android.view.View;

import com.vmloft.develop.library.example.R;
import com.vmloft.develop.library.example.common.AppActivity;
import com.vmloft.develop.library.tools.widget.VMTimerBtn;

import butterknife.BindView;
import butterknife.OnClick;
import com.vmloft.develop.library.tools.widget.record.VMRecordView;
import com.vmloft.develop.library.tools.widget.toast.VMToast;

/**
 * Created by lzan13 on 2017/4/1.
 * 测试录音控件
 */
public class CustomViewActivity extends AppActivity {

    @BindView(R.id.custom_btn_timer) VMTimerBtn mTimerBtn;
    @BindView(R.id.custom_record_view) VMRecordView mRecordView;

    @Override
    protected int layoutId() {
        return R.layout.activity_view_custom;
    }

    @Override
    protected void init() {
        mRecordView.setRecordListener(new VMRecordView.RecordListener() {
            @Override
            public void onStart() {
                VMToast.make(mActivity, "录音开始").done();
            }

            @Override
            public void onCancel() {
                VMToast.make(mActivity, "录音取消").error();
            }

            @Override
            public void onError(int code, String desc) {
                VMToast.make(mActivity, "录音失败 %d %s", code, desc).error();
            }

            @Override
            public void onComplete(String path, long time) {
                VMToast.make(mActivity, "录音完成 %s %d", path, time).done();
            }
        });
    }

    @OnClick({
        R.id.custom_btn_timer, R.id.btn_toast_1, R.id.btn_toast_2, R.id.btn_toast_3
    })
    public void onClick(View view) {
        switch (view.getId()) {
        case R.id.custom_btn_timer:
            mTimerBtn.startTimer();
            break;
        case R.id.btn_toast_1:
            VMToast.init(R.color.vm_black, R.drawable.vm_ic_camera, R.color.vm_amber);
            VMToast.make(mActivity, "测试自定义弹出 Toast 提醒功能，这是默认提醒样式！自定义颜色的").show();
            break;
        case R.id.btn_toast_2:
            VMToast.make(mActivity, "测试自定义弹出 Toast 提醒功能，这是完成提醒默认样式！绿色的").done();
            break;
        case R.id.btn_toast_3:
            VMToast.make(mActivity, "测试自定义弹出 Toast 提醒功能，这是错误提醒默认样式！红色的").error();
            break;
        }
    }
}

File: example/src/main/java/com/vmloft/develop/library/example/demo/custom/CustomViewActivity.kt
package com.vmloft.develop.library.example.demo.custom

import android.view.View

import com.alibaba.android.arouter.facade.annotation.Route

import com.vmloft.develop.library.example.R.color
import com.vmloft.develop.library.example.R.drawable
import com.vmloft.develop.library.example.R.layout
import com.vmloft.develop.library.example.base.AppActivity
import com.vmloft.develop.library.tools.widget.record.VMRecordView.RecordListener
import com.vmloft.develop.library.tools.widget.toast.VMToast

import kotlinx.android.synthetic.main.activity_view_custom.customRatioLayout
import kotlinx.android.synthetic.main.activity_view_custom.customRecordView
import kotlinx.android.synthetic.main.activity_view_custom.customTimerBtn

/**
 * Created by lzan13 on 2017/4/1.
 * 描述：测试录音控件
 */
@Route(path = "/VMLoft/CustomView")
class CustomViewActivity : AppActivity() {

    override fun layoutId(): Int {
        return layout.activity_view_custom
    }

    override fun initUI() {
        super.initUI()
        setTopTitle("自定义控件演示")
        customRecordView.setRecordListener(object : RecordListener() {
            override fun onStart() {
                VMToast.make(mActivity, "录音开始").done()
            }

            override fun onCancel() {
                VMToast.make(mActivity, "录音取消").error()
            }

            override fun onError(code: Int, desc: String?) {
                VMToast.make(mActivity, "录音失败 %d %s", code, desc).error()
            }

            override fun onComplete(path: String?, time: Long) {
                VMToast.make(mActivity, "录音完成 %s %d", path, time).done()
            }
        })
    }

    override fun initData() {}

    fun startTimer(view: View) {
        customTimerBtn.startTimer()
    }

    fun toast1(view: View) {
        VMToast.make(mActivity, "测试自定义弹出 Toast 提醒功能，这是默认提醒样式！自定义颜色的")
                .setBGColor(color.app_bg_dark)
                .setIcon(drawable.emoji_dog)
                .setMsgColor(color.app_title_light)
                .show()
    }

    fun toast2(view: View) {
        VMToast.make(mActivity, "测试自定义表情雨控件").done()
    }

    fun toast3(view: View) {
        VMToast.make(mActivity, "测试自定义弹出 Toast 提醒功能，这是错误提醒默认样式！红色的").error()
    }

    fun setRatioLayout(view: View) {
        customRatioLayout.setFollowWidth(false)
    }
}